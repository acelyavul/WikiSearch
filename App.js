import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import Home from './src/screens/Home.js';
import Results from './src/screens/Results';
import Theme from './src/Theme';

const Stack = createStackNavigator();

const App = () => {
  return (
    <NavigationContainer theme={Theme}>
      <Stack.Navigator initialRouteName="Home">
        <Stack.Screen
          name="Home"
          component={Home}
          options={{header: () => null}}
        />
        <Stack.Screen
          name="Results"
          component={Results}
          options={{title: 'Results'}}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App;
