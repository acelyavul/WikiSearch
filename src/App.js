import React from 'react';
import { RecoilRoot } from 'recoil';

import SearchContextProvider from './store/search-context';
import SearchBar from './components/SearchBar.js';

const App = () => {
  return (
    <RecoilRoot>
      <SearchContextProvider>
        <SearchBar />
      </SearchContextProvider>
    </RecoilRoot>
  );
};

export default App;
