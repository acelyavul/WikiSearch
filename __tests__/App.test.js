import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

jest.mock('../__mocks__/react-native-reanimated.js');
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

it('should render App properly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
