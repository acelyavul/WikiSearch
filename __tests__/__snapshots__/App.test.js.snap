// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App properly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": undefined,
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": undefined,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#f5f5f5",
                            "flex": 1,
                            "flexDirection": "column",
                            "justifyContent": "flex-start",
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "alignItems": "center",
                              "color": "#191970",
                              "fontSize": 34,
                              "fontWeight": "bold",
                              "marginVertical": 40,
                            }
                          }
                        >
                          WikiSearch
                        </Text>
                        <TextInput
                          onChangeText={[Function]}
                          placeholder="Type here..."
                          placeholderTextColor="#44449A"
                          style={
                            Object {
                              "borderColor": "#191970",
                              "borderRadius": 12,
                              "borderWidth": 3,
                              "fontSize": 20,
                              "marginBottom": 15,
                              "paddingLeft": 20,
                              "width": "80%",
                            }
                          }
                          testID="input"
                          value=""
                        />
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#191970",
                              "borderColor": "#191970",
                              "borderRadius": 12,
                              "borderWidth": 1,
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "marginTop": 10,
                              "opacity": 1,
                              "padding": 5,
                              "width": "30%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#fff",
                                "fontSize": 22,
                              }
                            }
                          >
                            Search
                          </Text>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                    <View
                      collapsable={false}
                      pointerEvents="box-none"
                      style={Object {}}
                    >
                      <View
                        accessibilityElementsHidden={false}
                        importantForAccessibility="auto"
                        onLayout={[Function]}
                        pointerEvents="box-none"
                        style={null}
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
